kolibri/plugins/video_mp4_render/assets/src/vue/index.vue

```diff
@@ -1,15 +1,223 @@
+              name: 'currentTimeDisplay',
+            },
+            {
+              name: 'timeDivider',
+            },
+            {
+              name: 'progressControl',
+            },
+            {
+              name: 'durationDisplay',
+            },
+            {
+              name: 'remainingTimeDisplay',
+            },
+            {
+              name: 'muteToggle',
+            },
+            {
+              name: 'VolumeBar',
+            },
+            {
+              name: 'playbackRateMenuButton',
+            },
+            {
+              name: 'captionsButton',
+            },
+            {
+              name: 'fullscreenToggle',
+            },
+          ],
+        },
+      },
+      () => {
+        const centerButtons = this.$els.videowrapper.childNodes[1];
+        const toggleButton = centerButtons
+          .getElementsByClassName('videotoggle')[0];
+        const replayButton = centerButtons
+          .getElementsByClassName('videoreplay')[0];
+        const forwardButton = centerButtons
+          .getElementsByClassName('videoforward')[0];
+
+        videojs(this.$els.video).on('useractive', () => {
+          toggleButton.classList.remove('userInactive');
+          replayButton.classList.remove('userInactive');
+          forwardButton.classList.remove('userInactive');
+        });
+        videojs(this.$els.video).on('userinactive', () => {
+          toggleButton.classList.add('userInactive');
+          replayButton.classList.add('userInactive');
+          forwardButton.classList.add('userInactive');
+        });
+
+        videojs(this.$els.video).on('play', () => {
+          this.setPlayState(true);
+        });
+
+        videojs(this.$els.video).on('pause', () => {
+          this.setPlayState(false);
+        });
+
+        videojs(this.$els.video).on('ended', () => {
+          this.setPlayState(false);
+        });
+      });
+    },
+    data: () => ({
+      videoUrl: 'http://clips.vorwaerts-gmbh.de/VfE_html5.mp4',
+      posterUrl: 'http://i3.ytimg.com/vi/kITJ6qH7jS0/hqdefault.jpg',
+      captionsUrl: require('./en.vtt'),
+      captionsLang: 'en',
+      captionsLabel: 'English',
+      playingState: false,
+    }),
+    methods: {
+      setPlayState(state) {
+        this.playState = state;
+        if (state === true) {
+          videoPlayer.$('.videotoggle').classList.add('videopaused');
+          videoPlayer.$('.videoreplay').classList.add('display');
+          videoPlayer.$('.videoforward').classList.add('display');
+        } else {
+          videoPlayer.$('.videotoggle').classList.remove('videopaused');
+          videoPlayer.$('.videoreplay').classList.remove('display');
+          videoPlayer.$('.videoforward').classList.remove('display');
+        }
+      },
+    },
+  };
 
 </script>
 
 
-<style lang="stylus" scoped></style>
+<style lang="stylus">
+
+  // Default videojs stylesheet
+  @import '../../../../../../node_modules/video.js/dist/video-js.css'
```

Comment:
mind adding a comment here about what problems we ran into trying to do this in a cleaner way?